### PR TITLE
Setup S3 connection with integrations

### PR DESCRIPTION
**Pull request by Swiddis**
### Description

Followup to #1052, this PR adds actual S3 integration functionality.

Work in progress -- Main querying pipeline is good, but need to do some work for the UI.

### Issues Resolved
N/A

### Check List
- [ ] New functionality includes testing.
  - [ ] All tests pass, including unit test, integration test and doctest
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
  - [ ] New functionality has user manual doc added
- [X] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: public/components/integrations/components/__tests__/__snapshots__/setup_integration.test.tsx.snap
Comment (derek-ho) [inline public/components/integrations/components/__tests__/__snapshots__/setup_integration.test.tsx.snap:53]: This looks wrong, can you take another look @Swiddis

Comment (Swiddis) [reply public/components/integrations/components/__tests__/__snapshots__/setup_integration.test.tsx.snap]: Woops, forgot to update a test -- fixed

**Review by YANG-DB — APPROVED** [review]

File: public/components/integrations/components/setup_integration.tsx
Comment (joshuali925) [inline public/components/integrations/components/setup_integration.tsx:132]: nit remove logs

Comment (kavithacm) [inline public/components/integrations/components/setup_integration.tsx:132]: Are these console comments necessary ? also in line no 151

File: public/components/integrations/components/setup_integration.tsx
Comment (joshuali925) [inline public/components/integrations/components/setup_integration.tsx:166]: i think this works but usually it's `error, loading, data`. If loading is true then display loading, else if error is not null then display error, otherwise data is useable. for example like https://swr.vercel.app/#overview

File: public/components/integrations/components/setup_integration.tsx
Comment (kavithacm) [inline public/components/integrations/components/setup_integration.tsx:335]: can we change this toast to more appropriate wording?

File: public/components/integrations/components/setup_integration.tsx
Comment (joshuali925) [inline public/components/integrations/components/setup_integration.tsx:218]: nit use `{...item}` to copy object

File: public/components/integrations/components/setup_integration.tsx
Comment (derek-ho) [reply public/components/integrations/components/setup_integration.tsx:335]: +1

File: public/components/integrations/components/__tests__/__snapshots__/setup_integration.test.tsx.snap
Comment (derek-ho) [inline public/components/integrations/components/__tests__/__snapshots__/setup_integration.test.tsx.snap:40]: Why are you removing type here

File: public/components/integrations/components/setup_integration.tsx
Comment (kavithacm) [inline public/components/integrations/components/setup_integration.tsx:326]: can we remove unwanted console log comments?

Comment (derek-ho) [inline public/components/integrations/components/setup_integration.tsx:326]: Please remove console logs

File: public/components/integrations/components/setup_integration.tsx
Comment (derek-ho) [inline public/components/integrations/components/setup_integration.tsx:143]: Is there better way than while true?

**Review by derek-ho — APPROVED** [review]
Approving to unblock, left a few comments

File: public/components/integrations/components/setup_integration.tsx
Comment (joshuali925) [inline public/components/integrations/components/setup_integration.tsx:300]: could you explain a bit what's an incrementer?

**Comment by derek-ho** [comment]
@Swiddis can you fix DCO

**Review by joshuali925 — APPROVED** [review]

File: public/components/integrations/components/setup_integration.tsx
Comment (Swiddis) [reply public/components/integrations/components/setup_integration.tsx:329]: What's more appropriate wording? We don't have any error information in the response that we can use

File: public/components/integrations/components/setup_integration.tsx
Comment (Swiddis) [reply public/components/integrations/components/setup_integration.tsx:298]: If I make a helper `const incrementProgress = () => setProgress(progress + 1)`, it doesn't correctly identify when progress updates, so it always computes `0 + 1`. Similar situation if I use `setProgress(progress + 1)` directly. The workaround is to manually track progress locally and use that instead of the actual progress variable.

File: public/components/integrations/components/setup_integration.tsx
Comment (Swiddis) [reply public/components/integrations/components/setup_integration.tsx:142]: Not sure what kind of polling timeout to put on it, I've heard some queries can take up to 10 mins and don't have any reference for when I should just tell the user that it isn't working

File: public/components/integrations/components/__tests__/__snapshots__/setup_integration.test.tsx.snap
Comment (Swiddis) [reply public/components/integrations/components/__tests__/__snapshots__/setup_integration.test.tsx.snap:41]: Stopped filling a default value when the integration isn't loaded yet -- We only use the type when we go to integration adding so it's safe to keep empty until it's loaded.

File: public/components/integrations/components/setup_integration.tsx
Comment (joshuali925) [reply public/components/integrations/components/setup_integration.tsx:295]: use `setProgress((progress) => progress + 1)`. it's a bit of counterintuitive issue in react where the value of progress is saved in the function

**Comment by Swiddis** [comment]
Dealt with small-scale PR feedback -- More large-scale changes will need to happen later. Just waiting on tests before merging.

**Pull request merged**
